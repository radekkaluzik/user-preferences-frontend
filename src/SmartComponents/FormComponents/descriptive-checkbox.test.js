import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DescriptiveCheckbox from './DescriptiveCheckbox';
import { Form, RendererContext } from '@data-driven-forms/react-form-renderer';

describe('Descriptive checkbox tests', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Form onSubmit={() => undefined}>
        {(props) => (
          <RendererContext.Provider
            value={{ formOptions: { internalRegisterField: () => undefined } }}
          >
            <DescriptiveCheckbox label="test label" clearedValue {...props} />
          </RendererContext.Provider>
        )}
      </Form>
    );
    expect(toJson(wrapper.find('DescriptiveCheckbox'))).toMatchSnapshot();
  });
});
