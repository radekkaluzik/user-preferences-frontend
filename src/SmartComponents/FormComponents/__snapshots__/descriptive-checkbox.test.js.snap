// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Descriptive checkbox tests should render correctly 1`] = `
<DescriptiveCheckbox
  __versions={
    Object {
      "final-form": "4.20.4",
      "react-final-form": "6.5.7",
    }
  }
  clearedValue={true}
  dirty={true}
  dirtyFields={
    Object {
      "": true,
    }
  }
  dirtyFieldsSinceLastSubmit={
    Object {
      "": true,
    }
  }
  dirtySinceLastSubmit={false}
  errors={Object {}}
  form={
    Object {
      "batch": [Function],
      "blur": [Function],
      "change": [Function],
      "destroyOnUnregister": false,
      "focus": [Function],
      "getFieldState": [Function],
      "getRegisteredFields": [Function],
      "getState": [Function],
      "initialize": [Function],
      "isValidationPaused": [Function],
      "mutators": Object {},
      "pauseValidation": [Function],
      "registerField": [Function],
      "reset": [Function],
      "resetFieldState": [Function],
      "restart": [Function],
      "resumeValidation": [Function],
      "setConfig": [Function],
      "submit": [Function],
      "subscribe": [Function],
    }
  }
  handleSubmit={[Function]}
  hasSubmitErrors={false}
  hasValidationErrors={false}
  invalid={false}
  isDanger={false}
  label="test label"
  modified={
    Object {
      "": false,
    }
  }
  modifiedSinceLastSubmit={false}
  name=""
  pristine={false}
  submitFailed={false}
  submitSucceeded={false}
  submitting={false}
  touched={
    Object {
      "": false,
    }
  }
  valid={true}
  validating={false}
  values={Object {}}
  visited={
    Object {
      "": false,
    }
  }
>
  <Checkbox
    checked={true}
    className="pref-c-descriptive-checkbox"
    data-type="descriptive-checkbox"
    id="descriptive-checkbox-"
    isChecked={true}
    isDisabled={false}
    isValid={true}
    label={
      <span
        className="pref-c-checkbox-label"
      >
        test label
      </span>
    }
    name=""
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    ouiaSafe={true}
    type="checkbox"
  >
    <div
      className="pf-c-check pref-c-descriptive-checkbox"
    >
      <input
        aria-invalid={false}
        checked={true}
        className="pf-c-check__input"
        data-ouia-component-id="OUIA-Generated-Checkbox-1"
        data-ouia-component-type="PF4/Checkbox"
        data-ouia-safe={true}
        data-type="descriptive-checkbox"
        disabled={false}
        id="descriptive-checkbox-"
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="descriptive-checkbox-"
      >
        <span
          className="pref-c-checkbox-label"
        >
          test label
        </span>
      </label>
    </div>
  </Checkbox>
</DescriptiveCheckbox>
`;
